File: Simple_Workshop_Projects/JavaProject/app/src/main/java/gr/sfhmmy/java/workshop/MainActivityViewModel.java
package gr.sfhmmy.java.workshop;

import androidx.lifecycle.ViewModel;

public class MainActivityViewModel extends ViewModel {

    public String calculatingGreeting(int age, String greeting) {
        if(age<5){greeting = "Hello Kid";}
        else if (age<18){greeting = "Hello " + greeting; }
        else if (age < 60) { greeting = "Hello Mr/Ms " + greeting; }
        else {
            greeting = "Hello Grandma/Grandpa";
        }
        return greeting;
    }
}

File: Simple_Workshop_Projects/KotlinProject/app/src/main/java/gr/sfhmmy/kotlin/workshop/MainActivityViewModel.kt
package gr.sfhmmy.kotlin.workshop

import androidx.lifecycle.ViewModel

class MainActivityViewModel : ViewModel() {
    fun calculatingGreeting(age: Int, greeting: String): String {
        var greeting = greeting
        greeting = if (age < 5) {
            "Hello Kid"
        } else if (age < 18) {
            "Hello $greeting"
        } else if (age < 60) {
            "Hello Mr/Ms $greeting"
        } else {
            "Hello Grandma/Grandpa"
        }
        return greeting
    }
}
